glass/fields.py

```diff
@@ -402,10 +406,11 @@ def effective_cls(  # type: ignore[no-untyped-def]
 
     # get the iterator over leading weight axes
     # auto-spectra do not repeat identical computations
-    if weights2 is weights1:
-        pairs = combinations_with_replacement(np.ndindex(shape1[1:]), 2)
-    else:
-        pairs = product(np.ndindex(shape1[1:]), np.ndindex(shape2[1:]))  # type: ignore[assignment]
+    pairs = (
+        combinations_with_replacement(np.ndindex(shape1[1:]), 2)
+        if weights2 is weights1
+        else product(np.ndindex(shape1[1:]), np.ndindex(shape2[1:]))
+    )
```

Comment:
I find this way harder to read, but I understand you don't want to manually type `pairs` here

Reply:
If it helps, they are faster too https://switowski.com/blog/for-loop-vs-list-comprehension/#why-is-list-comprehension-faster-than-a-for-loop